ext/drpttracer/drpttracer.h

```diff
@@ -0,0 +1,211 @@
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* DynamoRIO Intel PT Tracing Extension. */
+
+#ifndef _DRPTTRACER_H_
+#define _DRPTTRACER_H_ 1
+
+/**
+ * @file drpttracer.h
+ * @brief Header for DynamoRIO Intel PT Tracing Extension
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <inttypes.h>
+
+#ifndef IN
+#    define IN // nothing
+#endif
+#ifndef OUT
+#    define OUT // nothing
+#endif
+#ifndef INOUT
+#    define INOUT // nothing
+#endif
+
+/**
+ * The storage container type of a drpttracer's output buffer.
+ * The drpttracer_buf_t is used for the drpttracer to store the PT trace or Sideband data.
+ */
+typedef struct _drpttracer_buf_t {
+    void *buf;    /**< The buffer pointer */
+    int buf_size; /**< The buffer size */
+} drpttracer_buf_t;
+
+/**
+ * The type of PT trace's metadata. This type is used to initialize the PT config and PT
+ * sideband config.
+ */
+typedef struct __attribute__((__packed__)) _pt_metadata_t {
+    /** The CPU family. */
+    uint16_t cpu_family;
+    /** The CPU mode. */
+    uint8_t cpu_model;
+    /** The CPU stepping. */
+    uint8_t cpu_stepping;
+    /** perf_event_mmap_page.time_shift */
+    uint16_t time_shift;
+    /** perf_event_mmap_page.time_mult */
+    uint32_t time_mult;
+    /** perf_event_mmap_page.time_zero */
+    uint64_t time_zero;
+} pt_metadata_t;
+
+/***************************************************************************
+ * INIT
+ */
+
+DR_EXPORT
+/**
+ * Initializes the drpttracer extension.  Must be called prior to any of the
+ * other routines.  Can be called multiple times (by separate components,
+ * normally) but each call must be paired with a corresponding call to
+ * drpttracer_exit().
+ *
+ * \return whether successful.
+ */
+bool
+drpttracer_init(void);
+
+DR_EXPORT
+/**
+ * Cleans up the drpttracer extension.
+ */
+void
+drpttracer_exit(void);
+
+/***************************************************************************
+ * PT TRACING APIS
+ */
+
+/**
+ * Success code for each drpttracer operation.
+ */
+typedef enum {
+    /** Operation succeeded. */
+    DRPTTRACER_SUCCESS,
+    /** Operation failed. */
+    DRPTTRACER_ERROR,
+    /** Operation failed: invalid tracing handle. */
+    DRPTTRACER_ERROR_INVALID_TRACING_HANDLE,
+    /** Operation failed: invalid tracing mode. */
+    DRPTTRACER_ERROR_INVALID_TRACING_MODE,
+    /** Operation failed: failed to open perf event. */
+    DRPTTRACER_ERROR_FAILED_TO_OPEN_PERF_EVENT,
+    /** Operation failed: failed to create tracer handle. */
+    DRPTTRACER_ERROR_FAILED_TO_CREATE_PTTRACER_HANDLE,
+    /** Operation failed: failed to start tracing. */
+    DRPTTRACER_ERROR_FAILED_TO_START_TRACING,
+    /** Operation failed: failed to stop tracing. */
+    DRPTTRACER_ERROR_FAILED_TO_STOP_TRACING
+} drpttracer_status_t;
+
+/**
+ * The tracing modes for drpttracer_start_tracing().
+ */
+typedef enum {
+    /* drpttracer only traces in user space. */
```

Comment:
These 3 should all be Doxyen comments and should have cap + end punct

Reply:
Done.